roles/openshift_adm/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+- name: Ensure required variables are defined.
+  when: not lookup('ansible.builtin.vars', item)
+  ansible.builtin.fail:
```

Comment:
why not using `ansible.builtin.assert` ? That would avoid a loop, and the test would take only one action instead of "many" with the exposed loop.

Reply:
Thank you. It would be cleaner.